bin_sort/bin_sort.py

```diff
@@ -0,0 +1,94 @@
+"""
+Following is the implementation of bin sort algorithm.
+It will sort the input array with time complexity H(N + K)
+N being the size of array, K being the number of buckets.
+Space complexity is at worst O(NK).
+"""
+
+
+import math
+
+
+def insertion_sort(a):
+    """
+    :param a: list
+        List of objects with an order
+    :return: list
+        Sorted list with insertion sort algorithm
+    """
+
+    for i in range(1, len(a)):
+        current_value = a[i]
+        position = i
+
+        while position > 0 and a[position - 1] > current_value:
+            a[position] = a[position - 1]
+            position -= 1
+            a[position] = current_value
+
+    return a
+
+
+def get_buckets(a, bucket_size):
+    """
+    :param a: list
+        List of objects with an order
+    :param bucket_size: int
+        Size of buckets to fill
+    :return: list
+        List of buckets to sort array with
+    """
+
+    min_v, max_v = min(a), max(a)  # min and max values in array
+    bucket_count = int(math.floor((max_v - min_v) / bucket_size)) + 1
```

Comment:
@singhpratyush it would work on python3, but in python2 I need `bucket_count` to be an exact `int`: later on I call `range(bucket_count)` and python2 does not allow bucket_count a `float`